src/parser/base.py

```diff
@@ -40,32 +40,25 @@ def handle_execute(self, b_rate_functions, nb_rate_function):
         return result
 
     def safe_parsing(self, func, bank_id):
+        result = None
         try:
             result = func()
             if rate.is_empty(result):
                 raise ParseError("rates are empty")
-
-            return result
-
         except Exception as e:
             self.log.error({
                 "msg": str(e),
                 "country": self.country,
                 "bank_id": bank_id,
             })
-            return None
-
-    def remove_nones(self, rates):
-        ids = []
-
-        for i in rates:
-            if rates[i] is None:
-                ids.append(i)
-                continue
-
-        for i in ids:
-            del rates[i]
+        return result
 
+    @staticmethod
+    def remove_nones(rates):
+        """ Remove empty rates """
+        for rate in rates:
+            if rates[rate] is None:
+                del rates[rate]
```

Comment:
I believe it was not possible when initially this code was written on Python2.x. 
With Python3 looks like it works. Good improvement. Thanks.